torch/_inductor/pattern_matcher.py

```diff
@@ -253,6 +262,10 @@ def replace_by_example(
             replacement = trace_fn(
                 replacement_fn, torch.fx.map_arg(args, lambda arg: arg.meta["val"])  # type: ignore[arg-type]
             )
+            if len(self.nodes) > 0:
+                for n in replacement.graph.nodes:
+                    _transfer_meta(new_meta=n.meta, old_meta=self.nodes[0].meta)
```

Comment:
What if we have multiple nodes in `self.nodes`? Can we end up with more confusion downstream?

Reply:
Yes. I'd argue it's still better than having nothing though:
* In practice, it looks like very few replace_by_example patterns actually have multiple source nodes.
* The main case where you see this is in the (`add(z, mm(x, y)) -> addmm(z, x, y)` case.)
* original_aten may be confusing, but you'll also get other information like source location that will still be useful.

Do you have a suggestion on a better way to get this information

Reply:
Or alternatively, how about we only preserve metadata if self has exactly 1 node?

Reply:
> Or alternatively, how about we only preserve metadata if self has exactly 1 node?

IMO, this is potentially less confusing.

Reply:
I've changed the PR to do this (only preserve metadata if self has exactly 1 node)